File: elixir/leap/leap.exs
defmodule Year do
  def leap_year?(year) do
    cond do
      divisible_by?(year, 400) -> true
      divisible_by?(year, 100) -> false
      divisible_by?(year, 4) -> true
      true -> false
    end
  end

  defp divisible_by?(x, y), do: rem(x, y) === 0
end

File: erlang/leap/src/leap.erl
-module(leap).

-export([leap_year/1, test_version/0]).

leap_year(Year) ->
  if (Year rem 400) == 0 -> true
   ; (Year rem 100) == 0 -> false
   ; (Year rem 4)   == 0 -> true
   ; true                -> false
  end.

test_version() ->
  1.
